lib/android/index.js

```diff
@@ -39,10 +44,30 @@ class Android {
     });
   }
 
+  async _downloadDir(sourcePath, destinationPath) {
+    const files = await this.client.readdir(this.id, sourcePath);
+
+    for (var file of files) {
+      const fullSourcePath = `${sourcePath}/${file.name}`; 
+      const fullDestinationPath = path.join(destinationPath, file.name);
+
+      if (file.isFile()) {
+        await this._downloadFile(fullSourcePath, fullDestinationPath);
+      } else if (file.isDirectory()) {
+        await mkdirp(fullDestinationPath);
+        await this._downloadDir(fullSourcePath, fullDestinationPath);
```

Comment:
Something has to `await` all the promises.  I could collect them, I suppose, but this is simple (and close to deterministic -- if we wanted that, we could sort the directory listing) and doesn't require gathering.

Reply:
await-ing here is more correct. These need to run in this order.

Reply:
Same as above conversation, I don't think we need to await for downloadDir here and instead it can be done after the loop unless I'm missing something.  Either way, not a big deal, just a minor performance boost.